code/scripting/api/objs/oswpt.cpp

```diff
@@ -8,6 +8,89 @@
 #include "waypoint.h"
 #include "wing.h"
 #include "ship/ship.h"
+#include "mission/missionparse.h"
+
+#include "network/multi.h"
+#include "network/multimsgs.h"
+#include "network/multiutil.h"
+
+void object_ship_wing_point_team::serialize(lua_State* /*L*/, const scripting::ade_table_entry& /*tableEntry*/, const luacpp::LuaValue& value, ubyte* data, int& packet_size) {
+	object_ship_wing_point_team oswpt;
+	value.getValue(scripting::api::l_OSWPT.Get(&oswpt));
+	uint8_t oswpttype = static_cast<uint8_t>(oswpt.type);
+	ADD_DATA(oswpttype);
+	switch (oswpt.type) {
+	case oswpt_type::SHIP:
+	case oswpt_type::WAYPOINT:
+		ADD_USHORT(oswpt.objp->net_signature);
+		break;
+	case oswpt_type::PARSE_OBJECT:
+		ADD_USHORT(oswpt.ship_entry->p_objp->net_signature);
+		break;
+	case oswpt_type::WING:
+	case oswpt_type::WING_NOT_PRESENT: {
+		int wingidx = WING_INDEX(oswpt.wingp);
+		ADD_INT(wingidx);
```

Comment:
I directly added the wing index based on a discussion between us on the end of october: "Everybody _should_ have the same mission file and therefore the same wings.".
But I can change it to the wing name if desired, however that will also need to see the wing object itself be changed to a class with a custom serialize method, as the wing object currently is just an int that's directly serialized.

Reply:
.... let's get @notimaginative's opinion.  I'm probably being paranoid.

Reply:
Or we could use the net signature of the wing leader.  Since net signature is guaranteed to work, it should work.

Reply:
Looks to be perfectly safe to use the index. Check `send_wing_create_packet()` in multimsgs.cpp to see the same thing, including a comment from Volition concerning it's use. The wing index is also used in both the respawn and in-game join code.

Just be sure to sanity check it on the receiving end in case everything isn't sync'd up yet when the packet is processed.

Reply:
Added the same safety checks as for the wing create packet